src/azul/__init__.py

```diff
@@ -1165,9 +1165,45 @@ def is_lower_sandbox(self) -> bool:
         def is_unit_test(self):
             return self.name == self.test_name
 
+        def render(self) -> JSON:
+            cls = type(self)
+            rendered: JSON = {}
+            # Invoke each cached property getter and capture its value
+            for attribute in dir(cls):
+                descriptor = getattr(cls, attribute)
+                if isinstance(descriptor, cached_property):
+                    descriptor.fget(self)
+            return rendered
```

Comment:
This always returns an empty dictionary. I presume the intent was to update `rendered` inside the loop?